src/pages/workspaces/workspace/analysis/AzureComputeModal.js

```diff
@@ -0,0 +1,324 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h, label, p, span } from 'react-hyperscript-helpers'
+import { ButtonOutline, ButtonPrimary, IdContainer, Link, Select, spinnerOverlay, WarningTitle } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { NumberInput } from 'src/components/input'
+import { withModalDrawer } from 'src/components/ModalDrawer'
+import { InfoBox } from 'src/components/PopupTrigger'
+import TitleBar from 'src/components/TitleBar'
+import { Ajax } from 'src/libs/ajax'
+import {
+  azureMachineTypes, azureRegions, defaultAzureComputeConfig, defaultAzureDiskSize, defaultAzureMachineType, defaultAzureRegion, getMachineTypeLabel,
+  getRegionLabel
+} from 'src/libs/azure-utils'
+import colors from 'src/libs/colors'
+import { withErrorReporting, withErrorReportingInModal } from 'src/libs/error'
+import { useOnMount } from 'src/libs/react-utils'
+import {
+  computeStyles, getCurrentRuntime, getIsRuntimeBusy
+} from 'src/libs/runtime-utils'
+import * as Utils from 'src/libs/utils'
+
+
+// Change to true to enable a debugging panel (intended for dev mode only)
+const titleId = 'azure-compute-modal-title'
+
+export const AzureComputeModalBase = ({
+  onDismiss, onSuccess, workspace: { workspace: { namespace, name: workspaceName, workspaceId } }, runtimes
+}) => {
+  const [loading, setLoading] = useState(false)
+  const [viewMode, setViewMode] = useState(undefined)
+  const [currentRuntimeDetails, setCurrentRuntimeDetails] = useState(() => getCurrentRuntime(runtimes))
+
+  //TODO: revert before merging
+  var workspaceId = '1aa85f64-5717-4562-b3fc-2c963f66afa6'
+
+  const [computeConfig, setComputeConfig] = useState(defaultAzureComputeConfig)
+  const updateComputeConfig = (key, value) => setComputeConfig(_.set(key, value, computeConfig))
+
+  // Lifecycle
+  useOnMount(() => {
+    // Can't pass an async function into useEffect so we define the function in the body and then call it
+    const doUseOnMount = _.flow(
```

Comment:
I don't think pulling this out into a const is buying us much. Maybe lose it?
```suggestion
    _.flow(
```

Reply:
Hmm.. I was following the comment above. I don't the useEffect is compatible with async functions as the body. I see a lot of other examples of this pattern throughout the codebase.